build.sbt

```diff
@@ -1,17 +1,56 @@
-name := "circe-yaml"
-organization := "io.github.jeremyrsmith"
-description := "Library for converting between SnakeYAML's AST and circe's AST"
-version := "0.2.1"
-licenses += ("Apache-2.0", url("https://www.apache.org/licenses/LICENSE-2.0.html"))
+name := "circe-yaml-root"
+organization in ThisBuild := "io.github.jeremyrsmith"
+description in ThisBuild := "Library for converting between SnakeYAML's AST and circe's AST"
 
-scalaVersion := "2.11.8"
-crossScalaVersions := Seq("2.10.6", "2.11.8")
+val commonSettings = Seq(
+  version := "0.3.0",
+  scalaVersion := "2.11.8",
+  crossScalaVersions := Seq("2.10.6", "2.11.8"),
 
-libraryDependencies ++= Seq(
-  "io.circe" %% "circe-parser" % "0.5.0-M2",
-  "org.yaml" % "snakeyaml" % "1.17",
-  "org.scalatest" %% "scalatest" % "2.2.6" % "test"
+  libraryDependencies ++= Seq(
+    Libraries.circeCore
+  ) ++ Seq(
+    // Test-only dependencies
+    Libraries.circeTesting,
+    Libraries.discipline,
+    Libraries.scalaCheck,
+    Libraries.scalaTest
+  ).map(_ % Test),
+
+  bintrayRepository := "maven",
+  bintrayVcsUrl := Some("https://github.com/jeremyrsmith/circe-yaml"),
+
+  licenses += ("Apache-2.0", url("https://www.apache.org/licenses/LICENSE-2.0.html"))
 )
 
-bintrayRepository := "maven"
-bintrayVcsUrl := Some("https://github.com/jeremyrsmith/circe-yaml")
\ No newline at end of file
+lazy val core = (project in file("core"))
+  .settings(commonSettings)
+  .settings(
+    name := "circe-yaml-core",
+    libraryDependencies ++= Seq(
+      Libraries.snakeYaml
```

Comment:
Obviously I will need to remove this dependency by cleaning up the `YamlPrinterOptions` to avoid depending on SnakeYaml types.